controllers/web.js

```diff
@@ -0,0 +1,185 @@
+      .then(result => {
+        return res.status(200).json(result);
+      }, err => {
+        return res.status(400).json(err);
+      });
+  }
+
+
+
+
+  function onGet(req, res) {
+    let senderId = res.senderId = req.query.senderId;
+
+    if (senderId) {
+      return web.getUnread(res, senderId).then(result => {
+        return res.status(200).json(result);
+      }, err => {
+        return res.status(400).json(err);
+      });
+    } else {
+      senderId = uuid();
+      let but = [
+        new Button("web_url", `${config.server.url}${config.server.basePath}/authorize?state=${senderId}-web`,"se connecter")
+      ];
+      return sendResponse(null, senderId, new ButtonsListMessage("Pour commencer, connecter vous :)", but))
+        .then(() => res.status(200).json(senderId));
+    }
+  }
+
+
+  function onPost(req, res) {
+    let message = req.body.message;
+    let senderId = res.senderId = req.body.senderId;
+    let type = req.body.type;
+
+
+    //save user message to db first
+    let userMsg = {
+      message
+    };
+
+    if (type === "message") {
+      userMsg.origin = "user";
+    } else if (type === "postback") {
+      userMsg.origin = "user_postback";
+    }
+
+    return web
+      .pushToHistory(senderId, userMsg)
+      .then(() => {
+        if (type === "message") {
+          return messageReceived(res, senderId, message);
+        } else if(type === "postback") {
+          return postbackReceived(res, senderId, message);
+        }
+      })
+      .then((result) => {
+        if (result.responses) {
+          return sendResponse(res, senderId, result.responses);
+        } else {
+          return sendResponse(res, senderId, result);
+        }
+      })
+      .then((result) => {
+        return res.status(200).json(result);
+      })
+      .catch(err => {
+        res.logger.error(err);
+        return res.status(500).json(err);
+      });
+
+  }
+
+  function sendQuickResponses(res, senderId, responses) {
+    return Bluebird.mapSeries(responses, (response) => {
+      switch (response.type) {
+      case 0:
+      default:
+        let textResponse = response.speech.replace(/<(.*)\|+(.*)>/, "$1");
+        return Bluebird.resolve(textResponse);
+      }
+    });
+  }
+
+  function sendFeedback(res, senderId, intent, message) {
+    if (intent === "unknown") {
+      return;
+    }
+
+    message = message.length >= 1000 ? "TOOLONG" : message;
+
+    let buttons = [
+      new Button("postback", `FEEDBACK_MISUNDERSTOOD_${camelCase(intent)}_${message}`, "Mauvaise compréhension"),
+      new Button("postback", `FEEDBACK_BAD_${camelCase(intent)}_${message}`, "Non"),
+      new Button("postback", `FEEDBACK_GOOD_${camelCase(intent)}_${message}`, "Oui")
+    ];
+
+    return web.send(null, senderId, new ButtonsListMessage("Est-ce que cette réponse vous a aidé ?", buttons));
+  }
+
+  function sendResponse(res, senderId, response) {
+    return Bluebird.resolve(web.send(res, senderId, response));
```

Comment:
nope, `return web.send(res, senderId, response);` if you don't understand why, let me know